pkg/storage/etcd3/store.go

```diff
@@ -448,13 +448,6 @@ func (s *store) ttlOpts(ctx context.Context, ttl int64) ([]clientv3.OpOption, er
 	return []clientv3.OpOption{clientv3.WithLease(clientv3.LeaseID(lcr.ID))}, nil
 }
 
-func keyWithPrefix(prefix, key string) string {
-	if strings.HasPrefix(key, prefix) {
-		return key
```

Comment:
> But (assuming that the prefix is "/registry" as it is in production), and we would have a Resource call "registry", then this path would be triggered.

... and would be a bug, since you could then have a "registry" resource named "pods" and it would conflict with real pods